finger-exercises-react/react/ecmascript/02 - promises/index.js

```diff
@@ -1,9 +1,17 @@
 // Hint: use setInterval, create a new Promise and measure time with Date.now()
 
-export function delay() {
-
+export function delay(time) {
+  const initalTime = Date.now();
+  return new Promise((resolve, reject) => {
+    if (time > 500) reject(new Error('This time is too much !'));
+    setInterval(() => resolve(Date.now() - initalTime), time);
+  });
 }
 
-export function asyncDelay() {
-
+export function asyncDelay(time) {
```

Comment:
You have the same code that `delay`
You can just write:
```
export function asyncDelay(time) {
  return delay(time);
}
```

Reply:
You're right! I'm already on it.